presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -228,6 +230,15 @@ private static OrcPageSource createOrcPageSource(
                     .orElseThrow(() -> new IllegalArgumentException("Effective predicate is none"));
             List<OrcColumn> fileReadColumns = new ArrayList<>(columns.size());
             List<Type> fileReadTypes = new ArrayList<>(columns.size());
+            if (isFullAcid) {
```

Comment:
`fileReadColumns ` and `fileReadTypes ` are initialized as `new ArrayList<>(columns.size())`
they should be initialized with `+3` if `isFullAcid`